testcases/crypto/ec_func.c

```diff
@@ -162,6 +162,566 @@ _signVerifyParam signVerifyInput[] = {
+                        goto testcase_cleanup;
+                    }
+
+                    testcase_pass("*Derive shared secret ec=%lu, kdf=%lu, keylen=%lu, shared_data=%lu passed.", i,j,k,m);
+
+                    if (secret_keyA != CK_INVALID_HANDLE)
+                        funcs->C_DestroyObject(session, secret_keyA);
+                    if (secret_keyB != CK_INVALID_HANDLE)
+                        funcs->C_DestroyObject(session, secret_keyB);
+                }
+            }
+        }
+
+        if (publ_keyA != CK_INVALID_HANDLE)
+            funcs->C_DestroyObject(session, publ_keyA);
+        if (priv_keyA != CK_INVALID_HANDLE)
+            funcs->C_DestroyObject(session, priv_keyA);
+        if (priv_keyB != CK_INVALID_HANDLE)
+            funcs->C_DestroyObject(session, priv_keyB);
+        if (publ_keyB != CK_INVALID_HANDLE)
+            funcs->C_DestroyObject(session, publ_keyB);
+    }
+
+testcase_cleanup:
+    if (publ_keyA != CK_INVALID_HANDLE)
+        funcs->C_DestroyObject(session, publ_keyA);
+    if (priv_keyA != CK_INVALID_HANDLE)
+        funcs->C_DestroyObject(session, priv_keyA);
+    if (priv_keyB != CK_INVALID_HANDLE)
+        funcs->C_DestroyObject(session, priv_keyB);
+    if (publ_keyB != CK_INVALID_HANDLE)
+        funcs->C_DestroyObject(session, publ_keyB);
+    if (secret_keyA != CK_INVALID_HANDLE)
+        funcs->C_DestroyObject(session, secret_keyA);
+    if (secret_keyB != CK_INVALID_HANDLE)
+        funcs->C_DestroyObject(session, secret_keyB);
+
+    testcase_user_logout();
+    testcase_close_session();
+
+	return rc;
+} /* end run_DeriveECDHKey() */
+
+/*
+ * Run some ECDH known answer tests.
+ */
+CK_RV run_DeriveECDHKeyKAT()
+{
+    CK_SESSION_HANDLE session;
+    CK_MECHANISM mech;
+    CK_FLAGS flags;
+    CK_OBJECT_HANDLE publ_keyA = CK_INVALID_HANDLE, priv_keyA = CK_INVALID_HANDLE;
+    CK_OBJECT_HANDLE publ_keyB = CK_INVALID_HANDLE, priv_keyB = CK_INVALID_HANDLE;
+    CK_OBJECT_HANDLE secret_keyA = CK_INVALID_HANDLE, secret_keyB = CK_INVALID_HANDLE;
+    CK_BYTE user_pin[PKCS11_MAX_PIN_LEN];
+    CK_KEY_TYPE secret_key_type = CKK_GENERIC_SECRET;
+    CK_OBJECT_CLASS class = CKO_SECRET_KEY;
+    CK_ECDH1_DERIVE_PARAMS ecdh_parmA, ecdh_parmB;
+    CK_ULONG user_pin_len;
+    CK_RV rc = CKR_OK;
+    CK_BYTE secretA_value[1000]; // enough space for key lengths in ecdh_tv[]
+    CK_BYTE secretB_value[1000];
+
+    testcase_begin("starting run_DeriveECDHKeyKAT...");
+    testcase_rw_session();
+    testcase_user_login();
+
+    if (!mech_supported(SLOT_ID, CKM_ECDH1_DERIVE)) {
+        testcase_skip("Slot %u doesn't support CKM_ECDH1_DERIVE\n", (unsigned int) SLOT_ID);
+        goto testcase_cleanup;
+    }
+
+    if (is_ep11_token(SLOT_ID)) {
+        testcase_skip("This testcase uses KDFs, currently not supported by ep11token.\n");
+        goto testcase_cleanup;
+    }
+
+    for (int i=0; i<ECDH_TV_NUM; i++) {
```

Comment:
Do not declare int i inside the for, please do it outside. This is causing Travis CI build to fail.